subsys/bluetooth/mesh/cfg_srv.c

```diff
@@ -2039,10 +2039,17 @@ static int mod_app_get(struct bt_mesh_model *model,
 	if (mod) {
 		int i;
 
-		for (i = 0; i < mod->keys_cnt; i++) {
-			if (mod->keys[i] != BT_MESH_KEY_UNUSED) {
-				net_buf_simple_add_le16(&msg, mod->keys[i]);
+		for (i = 0; i < mod->keys_cnt - 1; i += 2) {
+			if (mod->keys[i] == BT_MESH_KEY_UNUSED &&
+			    mod->keys[i + 1] == BT_MESH_KEY_UNUSED) {
+				break;
```

Comment:
Hm... What will happen if I fill list of keys fully but then remove two of them those are located consequently and then request model list?
It would be nice to have covered this case in bsim or unit tests.

Reply:
Hmm... I will investigate this. If that is an issue I guess we had more than one bug here.